tests/cli/mirroring_smoke_tests/e2e/fluvio-core.bats

```diff
@@ -0,0 +1,177 @@
+    assert_output ""
+    assert_success
+
+    run jq -r .home.id remote2.json 
+    assert_output "$HOME_NAME"
+    assert_success
+
+    run jq -r .home.remoteId remote2.json 
+    assert_output "$REMOTE_NAME_2"
+    assert_success
+
+    run jq -r .home.publicEndpoint remote2.json 
+    assert_output --partial ":30003"
+    assert_success
+}
+
+@test "Can create a mirror topic with the remote clusters" {
+    echo "[\"$REMOTE_NAME\",\"$REMOTE_NAME_2\"]" > remotes_devices.json
+    run timeout 15s "$FLUVIO_BIN" topic create "$TOPIC_NAME" --mirror-apply remotes_devices.json
+
+    assert_output "topic \"$TOPIC_NAME\" created"
+    assert_success
+}
+
+@test "Can switch to remote cluster 1" {
+    run timeout 15s "$FLUVIO_BIN" profile switch "$REMOTE_PROFILE"
+    assert_output ""
+    assert_success
+}
+
+@test "Can connect to the home cluster from remote 1" {
+    run timeout 15s "$FLUVIO_BIN" home connect --file remote.json
+
+    assert_output "connecting with \"$HOME_NAME\" cluster"
+    assert_success
+}
+
+@test "Can produce message to mirror topic from remote 1" {
+    sleep 5
+    run bash -c 'echo 1 | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo a | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo 2 | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo b | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+}
+
+@test "Can switch to remote cluster 2" {
+    run timeout 15s "$FLUVIO_BIN" profile switch "$REMOTE_PROFILE_2"
+    assert_output ""
+    assert_success
+}
+
+@test "Can connect to the home cluster from remote 2" {
+    run timeout 15s "$FLUVIO_BIN" home connect --file remote2.json
+
+    assert_output "connecting with \"$HOME_NAME\" cluster"
+    assert_success
+}
+
+@test "Can produce message to mirror topic" {
+    sleep 5
+    run bash -c 'echo 9 | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo z | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo 8 | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+    run bash -c 'echo y | timeout 15s "$FLUVIO_BIN" produce "$TOPIC_NAME"'
+    assert_success
+}
+
+@test "Can switch back to home cluster" {
+    run timeout 15s "$FLUVIO_BIN" profile switch "$HOME_PROFILE"
+    assert_output ""
+    assert_success
+}
+
+#TODO: use --mirror argument when it is available
+@test "Can consume message from mirror topic produced from remote 1" {
+    sleep 5
+    run timeout 15s "$FLUVIO_BIN" consume "$TOPIC_NAME" -p 0 -B -d
+
+    assert_output 1$'\n'a$'\n'2$'\n'b
+    assert_success
+}
+
+@test "Can consume message from mirror topic produced from remote 2" {
+    sleep 5
+    run timeout 15s "$FLUVIO_BIN" consume "$TOPIC_NAME" -p 1 -B -d
+
+    assert_output 9$'\n'z$'\n'8$'\n'y
+    assert_success
+}
```

Comment:
need couple of more test cases:
- produce message on remote 1 before connecting.  
- disconnect home from remote 2.  then produce 2nd message on remote 2

Reply:
I added a test for the first one, but we don't have disconnect command to have the second. 
But I added a TODO comment 😅

Reply:
you can simulate same condition by:
- shutdown remote cluster
- restart remote cluster
- produce message on remote.

probably need command on remote cluster to shutdown connection in the future.

Reply:
I tried, unfortunately, `shutdown` seems to have the same problem as `cluster delete` of deleting all executables by name. Also, resume does not work after it when it's running two clusters for some reason, with one works.

Reply:
Ok. let's create issue for this.  this is critical path to cover.  thanks for trying